scripts/merge-script/config.ts

```diff
@@ -0,0 +1,74 @@
+import {dirname, resolve} from 'path';
+
+/**
+ * Target labels represent Github pull requests labels. These labels instruct the merge
+ * script into which branches a given pull request should be merged to.
+ */
+export interface TargetLabel {
+  /** Pattern that matches the given target label. */
+  pattern: RegExp|string;
+  /**
+   * List of branches a pull request with this target label should be merged into.
+   * Can also be wrapped in a function that accepts the target branch specified in the
+   * Github Web UI. This is useful for supporting labels like `target: development-branch`.
+   */
+  branches: string[]|((githubTargetBranch: string) => string[]);
+  /** Optional SHA which pull requests with this label should be based on top of. */
+  requiredBaseSha?: string;
+}
+
+/** Configuration for the merge script. */
+export interface Config {
+  /** Relative path (based on the config location) to the project root. */
+  projectRoot: string;
+  /** Configuration for the upstream repository. */
+  repository: {user: string; name: string; useHttps?: boolean};
+  /** List of target labels. */
+  labels: TargetLabel[];
+  /** Pattern that matches labels which imply a signed CLA. */
+  claSignedLabel: string|RegExp;
+  /** Pattern that matches label which imply a merge ready pull request. */
+  mergeReadyLabel: string|RegExp;
+}
+
+/** Reads and validates the configuration file at the specified location. */
+export function readAndValidateConfig(filePath: string): {config?: Config, errors?: string[]} {
+  try {
+    const config = require(filePath)() as Config;
+    const errors = validateConfig(config);
+    if (errors.length) {
+      return {errors};
+    }
+    // Resolves the project root path to an absolute path based on the
+    // config file location.
+    config.projectRoot = resolve(dirname(filePath), config.projectRoot);
+    return {config};
+  } catch (e) {
+    return {errors: [`File could not be loaded. Error: ${e.message}`]};
+  }
+}
+
+/** Validates the specified configuration. Returns a list of failure messages. */
+function validateConfig(config: Config): string[] {
+  const errors: string[] = [];
+  if (!config.projectRoot) {
+    errors.push('Missing project root.');
+  }
+  if (!config.labels) {
+    errors.push('No label configuration.');
+  } else if (!Array.isArray(config.labels)) {
+    errors.push('Label configuration needs to be an array.');
+  }
+  if (!config.repository) {
+    errors.push('No repository is configured.');
+  } else if (!config.repository.user || !config.repository.name) {
+    errors.push('Repository configuration needs to specify a `user` and repository `name`.');
+  }
+  if (!config.claSignedLabel) {
+    errors.push('No CLA signed label configured.');
+  }
+  if (!config.mergeReadyLabel) {
+    errors.push('No merge ready label configured.');
+  }
+  return errors;
+}
```

Comment:
The config parsing will be easy to integrate w` `getDevInfraConfig()` if we eventually move it to `/dev-infra`.